web/src/features/AppConfig/components/AppConfig.tsx

```diff
@@ -495,12 +495,22 @@ class AppConfig extends Component<Props, State> {
     const hasRequiredValidationError =
       showErrorsForEmptyValues &&
       Boolean(
-        groups.find((group) =>
-          Boolean(group?.items.find((item) => item.required && !item.value))
+        groups?.find(
+          (group) =>
+            group?.when !== "false" &&
```

Comment:
we have to check when at the group and item level. and hidden at just the item level per https://docs.replicated.com/reference/custom-resource-config#when